src/emcpy/plots/scatter.py

```diff
@@ -0,0 +1,104 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.colors import Normalize
+from scipy.interpolate import interpn
+from emcpy.stats import get_linear_regression
+
+__all__ = ['scatter']
+
+
+def _density_scatter(x, y, ax=None, sort=True, bins=20, **kwargs):
+    """
+    Creates scatter plot colored by 2d histogram
+    """
+    data, x_e, y_e = np.histogram2d(x, y, bins=bins, density=True)
+    z = interpn((0.5*(x_e[1:] + x_e[:-1]), 0.5*(y_e[1:]+y_e[:-1])),
+                data, np.vstack([x, y]).T, method="splinef2d",
+                bounds_error=False)
+    # To be sure to plot all data
+    z[np.where(np.isnan(z))] = 0.0
+    # Sort the points by density, so that the densest points are plotted last
+    if sort:
+        idx = z.argsort()
+        x, y, z = x[idx], y[idx], z[idx]
+    ax.scatter(x, y, c=z, **kwargs)
+    norm = Normalize(vmin=np.min(z), vmax=np.max(z))
+
+    return ax
+
+
+def _gen_scatter(x, y, plotopts):
+    """
+    Generate scatter plot from x and y.
+    """
+    fig = plt.figure(figsize=(8, 6))
+    ax = fig.add_subplot(111)
+
+    if plotopts['density']:
+        ax = _density_scatter(x, y, ax=ax, bins=[100, 100],
+                              s=4, cmap=plotopts['cmap'])
+    else:
+        plt.scatter(x, y, s=4, color=plotopts['color'],
+                    label=f'n={np.count_nonzero(~np.isnan(x))}')
+
+    if plotopts['linear_regression']:
+        y_pred, r_sq, intercept, slope = get_linear_regression(x, y)
+        label = f"y = {slope:.4f}x + {intercept:.4f}\nR\u00b2 : {r_sq:.4f}"
+        ax.plot(x, y_pred, color=plotopts['r color'], linewidth=1, label=label)
+
+    plt.grid(plotopts['grid'])
+    plt.legend(loc='upper left', fontsize=11)
+    plt.title(plotopts['title'], loc='left')
+    plt.title(plotopts['cycle'], loc='right', fontweight='semibold')
```

Comment:
can we change `cycle` to `time`? What if it is an average from July 1 through July 31 (for example)?

Reply:
Or even better, can we have something like `secondary title` (please come up with a better name) to further generalize this? Or will it always be a title and a time?

Reply:
My personal preference, I think the time should be included. The first title should include information about what your plotting and the type of data and the second title should be time. I can change it to like `datetime` or something, but should be included. Right now, if it is `None` it just won't show up. However, if you just want one title and you can include whatever you want including info, dates, sensors, etc. and it will be in the middle of the figure that's fine.

Reply:
I'm fine with defaulting to the right title being 'time', I just didn't like the key `cycle` as it doesn't apply if it is an average or a scatter of all points in a month, etc.. Since it is just a string being passed, I think a generic name like `time` here works. 

That way it can say:
```
Valid: 2021072600
Init: 2021072600 Valid: 2021072606
2021072600 - 2021072700
Friday, July 30, 2021
```
etc.